pkg/controller/replicaset/replica_set_test.go

```diff
@@ -693,7 +702,52 @@ func TestUpdatePods(t *testing.T) {
 				t.Errorf("Expected keys %#v got %v", expected, got)
 			}
 		case <-time.After(wait.ForeverTestTimeout):
-			t.Errorf("Expected update notifications for replica sets within 100ms each")
+			t.Errorf("Expected update notifications for replica sets")
+		}
+	}
+
+	// case 2: Remove ControllerRef (orphan). Expect to sync both former owner and
+	// any label-matching RS.
+	pod1 = newPodList(informers.Core().V1().Pods().Informer().GetIndexer(), 1, v1.PodRunning, labelMap1, testRSSpec1, "pod").Items[0]
+	pod1.ResourceVersion = "1"
+	pod1.Labels = labelMap2
+	pod1.OwnerReferences = []metav1.OwnerReference{controllerRef1}
+	pod2 = pod1
+	pod2.OwnerReferences = nil
+	pod2.ResourceVersion = "2"
+	manager.updatePod(&pod1, &pod2)
+	expected = sets.NewString(testRSSpec1.Name, testRSSpec2.Name)
+	for _, name := range expected.List() {
+		t.Logf("Expecting update for %+v", name)
+		select {
+		case got := <-received:
+			if !expected.Has(got) {
+				t.Errorf("Expected keys %#v got %v", expected, got)
+			}
+		case <-time.After(wait.ForeverTestTimeout):
+			t.Errorf("Expected update notifications for replica sets")
+		}
+	}
+
+	// case 4: Keep ControllerRef, change labels. Expect to sync owning RS.
+	pod1 = newPodList(informers.Core().V1().Pods().Informer().GetIndexer(), 1, v1.PodRunning, labelMap1, testRSSpec1, "pod").Items[0]
+	pod1.ResourceVersion = "1"
+	pod1.Labels = labelMap1
+	pod1.OwnerReferences = []metav1.OwnerReference{controllerRef2}
+	pod2 = pod1
+	pod2.Labels = labelMap2
+	pod2.ResourceVersion = "2"
+	manager.updatePod(&pod1, &pod2)
```

Comment:
if the test case is changing label from labelMap2 to labelMap1 (i.e., changing from match controller's selector to not match), is the expected result the same?